### ISSUE TITLE
stale-closure — Error tier minted from a partial proof (name-matched registrar, may self-write)

### ISSUE THREAD

**Issue by rboudrouss** [rboudrouss/reactant-analyzer#142]
## Shape

`stale-closure` mints its Error tier from a proof that covers one conjunct of the claim. The token is `must_on_all_paths` over the registration block (`stale_closure.rs`, `reach_proof`); the other two conjuncts are may-facts:

1. **The registrar is a name match.** `subscribe` / `on` / `addListener` are `Firing::Repeating` in the registrar table, so a custom method with one of those names reaches Error. #42 closed this heuristic as wontfix on the stated ground that it is "Warning-ceiling by construction, it can never produce an Error", and `docs/limitations.md` says "A false positive never carries an Error". Both are false today.

   ```js
   const registry = { on(name, fn) { fn(); } };
   useEffect(() => { registry.on('init', () => setN(n + 1)); }, []);
   // error stale-closure … the state can never advance past its first update
   ```

2. **The self-write is any write.** `self_write` comes from `collect_setter_calls_with_extra`, which collects conditional and nested calls alike:

   ```js
   useEffect(() => {
     const id = setInterval(() => { if (Math.random() > 2) setN(n + 1); }, 1000);
     return () => clearInterval(id);
   }, []);
   // error stale-closure
   ```

Same class as #95: an FP reaching Error breaks the severity contract.

## Root cause

`Certified<OnAllPaths>` proves the registration is reached, not the finding. The typestate guarantees an Error holds *a* proof, not that the proof covers the claim.

## Fix direction

One must-primitive per Error claim, as `must_frozen_seed` and `must_effect_cycle` already do: `must_stale_capture` in `api/query.rs` mints only when the registrar is `Repeating` with a proven timing (the name-heuristic rows are exactly the `Timing::Unknown` ones), the effect is mount-only, the registration is on all paths of the effect body, and the self-write is on all paths of the callback. Anything short of that stays Warning.

**Comment by rboudrouss** [comment]
Fixed in 307f6c7: `must_stale_capture` certifies the whole claim (proven-timing repeating registrar, mount-only effect, registration and write back on every path). Corpus: no stale-closure finding moved (both were already Warnings); baseline change in a7387ba is an unrelated naming merge.
